src/gc/gc.cpp

```diff
@@ -29036,39 +29313,44 @@ void gc_heap::mark_through_cards_for_segments (card_fn fn, BOOL relocating)
                                  dprintf (4, ("<%Ix>:%Ix", (size_t)poo, (size_t)*poo));
                                  if (card_of ((uint8_t*)poo) > card)
                                  {
-                                    BOOL passed_end_card_p  = card_transition ((uint8_t*)poo, end,
+                                     BOOL passed_end_card_p  = card_transition ((uint8_t*)poo, end,
                                             card_word_end,
                                             cg_pointers_found, 
                                             n_eph, n_card_set,
                                             card, end_card,
                                             foundp, start_address,
                                             limit, total_cards_cleared);
 
+                                     if (passed_end_card_p && !foundp)
+                                     {
+                                         foundp = FIND_NEXT_CHUNK(card_mark_enumerator, seg, n_card_set, start_address, limit, card, end_card, card_word_end);
```

Comment:
>foundp = FIND_NEXT_CHUNK(card_mark_enumerator, seg, n_card_set, start_address, limit, card, end_card, card_word_end); [](start = 41, length = 117)

if you made find_next_chunk part of card_transition you could also get rid of this macro

Reply:
Agreed.

Reply:
Done.